Shopping cart example

hedgerh opened the pull request:
For some reason the product images can't be found.  Guessing it is some webpack config issue, if anyone has any suggestions.

Other than that, it should be good to go.

As per issue https://github.com/rackt/redux/issues/823

hedgerh commented:
@gaearon

hedgerh commented:
Going to clean up the linting errors later tonight.  Still not sure about why product images are 404ing.

gaearon commented:
>For some reason the product images can't be found.

No need for images—let's keep it simple.

npverni commented on examples/shopping-cart/components/Cart.js:
```diff
@@ -0,0 +1,61 @@
+    });
+
+    return ( < div className = "cart uk-panel uk-panel-box uk-panel-box-primary" >
+      < div className = "uk-badge uk-margin-bottom" > Your Cart < /div> < div className = "uk-margin-small-bottom" > {
```
Is there a reason for the space before the tag names here?

hedgerh replied:
@npverni Weird side effect of the prettifying tool I used to convert from 4 space -> 2 space indentation.  Fixed!  Still need to work on the linting issues.

hedgerh commented:
This is ready for review. :octopus:

omnidan commented on examples/shopping-cart/actions/index.js:
```diff
@@ -0,0 +1,50 @@
+        cart
+      });
+      // Replace the line above with line below to rollback on failure:
+      // dispatch({ type: CHECKOUT_FAILURE, cart });
```
in the comment, this should be `types.CHECKOUT_FAILURE`

omnidan commented:
@hedgerh small typo in the comment, otherwise it looks good to me :ok_hand:

omnidan commented on examples/shopping-cart/components/Cart.js:
```diff
@@ -0,0 +1,35 @@
+      <div>Please add some products to cart.</div> :
+      products.map(product =>
+        <Product key={product.id}>
+          {product.title} - &euro;{product.price} x {product.quantity}
```
Any reason why you don't pass the title, price and quantity to `<Product>`?

omnidan commented on examples/shopping-cart/components/ProductItem.js:
```diff
@@ -0,0 +1,27 @@
+
+    return (
+      <div>
+        <h4>{product.title} - &euro;{product.price}</h4>
```
If you do what I said above, you could re-use `<Product>` here

hedgerh commented:
Thanks for taking the time to look it over, Daniel. And nice catch! I am
too full of food to get out of bed, but I'll fix it tomorrow :D
On Oct 25, 2015 5:23 AM, "Daniel Bugl" <notifications@github.com> wrote:

> @hedgerh <https://github.com/hedgerh> small typo in the comment,
> otherwise it looks good to me [image: :ok_hand:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rackt/redux/pull/824#issuecomment-150904669>.
>

hedgerh commented:
@omnidan: Made the corrections to the comment typo and I'm passing the title/quantity/price to the Product component as props.  Also changed ProductItem to use the Product component.  Added a little check in Product's render method to not render the 'x' if quantity wasn't passed.

omnidan commented:
@hedgerh looks good now :+1:

gaearon commented:
I'd appreciate a few fixes:

**Items need paddings.**

Right now they're way to close.

<img width="222" alt="screen shot 2015-10-28 at 15 11 22" src="https://cloud.githubusercontent.com/assets/810438/10788103/32b5481c-7d86-11e5-8ae4-75cc719a75b9.png">

**No need to mention Flux :-)**

Let's change “Flux Shop Demo (Redux)” title to “Shopping Cart”.
